extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -313,23 +314,58 @@ export async function promptForBindingType(funcName?: string): Promise<BindingTy
 /**
  * Prompts the user to enter object name for the SQL query
  * @param bindingType Type of SQL Binding
+ * @param connectionInfo (optional) connection info from the selected connection profile
+ * @returns the object name from user's input or menu choice
  */
-export async function promptForObjectName(bindingType: BindingType): Promise<string | undefined> {
-	return vscode.window.showInputBox({
-		prompt: bindingType === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
-		placeHolder: constants.placeHolderObject,
-		validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
-		ignoreFocusOut: true
-	});
+export async function promptForObjectName(bindingType: BindingType, connectionInfo?: IConnectionInfo): Promise<string | undefined> {
+	// show the connection string methods (user input and connection profile options)
+	let connectionURI: string | undefined;
+	let selectedDatabase: string | undefined;
+
+	while (true) {
+		// TODO create path to solve for selectView (first need to support views as well)
+		// Prompt user to select a table based on connection profile and selected database
+		const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+
+		if (!connectionInfo) {
```

Comment:
So one scenario that a user can choose is to enter a connectionString manually (when they did not include connectionInfo for promptAndUpdateConnectionStringSetting: https://github.com/microsoft/azuredatastudio/pull/19358/files/ea7ce50a8ff6d4d6ef25dfff76e816e608ea1527#diff-942c301818b2962d582b470ff15d00da98787e95e3bc98a95f79de6bae4be102R477

So if a user chooses that path it would mean they would result in an undefined connectionInfo. I believe this was the only scenario that would be covered by this. But since this could risk in allowing user to choose a different connection profile then what they choose for their connectionString then I could see that as the only risk in terms of it not working. 

Would you suggest instead of this if we have an undefined connectionInfo then we just prompt user to manually type in the objectName instead?

Reply:
Why not pass in the connection info for the connection string they entered instead in that case?

Reply:
I do that for all the scenarios after `promptAndUpdateConnectionStringSetting` but in this case, if a user uses adds SQL binding using quickpick. Since they have the option to enter a connection string manually we would return a connectionInfo as undefined from `promptAndUpdateConnectionStringSetting`.

Reply:
I'm saying if they enter a connection string manually then it seems like the best behavior would be to parse that into a connection info object and pass that in here. Is that something we could be doing?

Reply:
So introducing further complexity of parsing is something I believe we should take post-build release, ie June Release. I noted in the figma diagram that this would be a crawl experience and parsing the connection string the user manually enters and then setting it properly would be something we could address later (but I am trying to mitigate the risk here - without adding more to this already new feature). 

Do you think this is a good plan?

@chlafreniere for thoughts as well. 
Context - having the user manually type in the object if they manually entered the connection string (since we do not have the connectionInfo object to prompt for databases/tables).

Reply:
Alright, definitely want to avoid complexity this late so as long as we're not regressing anything that's fine.